cibyl/sources/elasticsearch/api.py

```diff
@@ -355,6 +364,125 @@ def get_deployment(self, **kwargs):
 
         return AttributeDictValue("jobs", attr_type=Job, value=job_objects)
 
+    @speed_index({'base': 3})
+    def get_tests(self, **kwargs):
+        """
+            Get tests for a elasticsearch job.
+
+            :returns: container of jobs with the last completed build
+            (if any) and the tests
+            :rtype: :class:`AttributeDictValue`
+        """
+        jobs_found = self.get_jobs(**kwargs)
+        query_body = {
+            "query": {
+                "bool": {
+                    "must": [
+                        {},
+                        {
+                            "bool": {
+                              "should": [
+                                {}
+                              ]
+                            }
+                        },
+                        {
+                            "exists": {
+                                "field": "test_status"
+                            }
+                        },
+                        {
+                            "exists": {
+                                "field": "test_name"
+                            }
+                        }
+                    ]
+                }
+            },
+            "sort": [{"timestamp.keyword": {"order": "desc"}}]
+        }
+
+        build_numbers = []
+        if 'build_number' in kwargs:
+            build_numbers = [status.upper()
+                             for status in
+                             kwargs.get('build_number').value]
+
+        results = []
+        hits = []
+        for job in jobs_found:
+            query_body['query']['bool']['must'][0] = {
+                "match": {
+                    "job_name.keyword": f"{job}"
+                }
+            }
+            for build_number in build_numbers:
+                query_body['query']['bool']['must'][1]['bool']['should'] = {
```

Comment:
Be careful using index access so freely. If just one of these indexes does not exist, then you will get an AttributeError out of here.

Reply:
There will no be problem with this, I replace always the same index that is a dictionary that contains a DSL `match` directive.

Reply:
Ok, if you know it will not bring trouble you can close this one down.